### ISSUE TITLE
`cp` does not work for directories

### ISSUE THREAD

**Issue by jayschwa** [JuliaLang/julia#10434]
``` julia
mkdir("foo")
cp("foo", "bar")
```
Expected result: `bar` is a directory.
Actual result: `bar` is an empty file.

Julia 0.3.6 on Windows 8.1 64-bit

**Comment by ivarne** [comment]
It doesn't work in OSX either (`0.3.3` and `0.4.0-dev+3699`)

```julia
ulia> cp("foo", "bar")
ERROR: sendfile: operation not supported on socket (ENOTSUP)
 in sendfile at fs.jl:142
 in cp at file.jl:73
```
The result is that `bar` is an empty file.

**Comment by jiahao** [comment]
IIRC neither the UNIX `cp` command nor the analogous Windows `copy` command allow the copying of directories. One has to issue other flags/commands like `cp -R` or `xcopy /s`.

Arguably, the correct behavior of `cp()` should be to do nothing if the source is not a file, and possibly raise a warning or error. (So the current behavior of creating an empty file is still wrong.)

It would be useful to have a recursive `cp()`, possibly with an optional keyword argument, or another function altogether. For comparison, Python has [`shutil.copy()`](https://docs.python.org/2/library/shutil.html#shutil.copy) and [`shutil.copytree()`](https://docs.python.org/2/library/shutil.html#shutil.copytree).

**Comment by simonster** [comment]
I'd vote for a `recursive` kwarg to match `rm`.

**Comment by garborg** [comment]
FWIW, @simonster's suggestion is how I assumed it worked already.

**Comment by tkelman** [comment]
closed by #10448
